skrf/media/media.py

```diff
@@ -1969,3 +1961,67 @@ def parse_z0(s: str) -> NumberLike:
     else:
         raise ValueError('couldnt parse z0 string')
     return out
+
+def get_z0_load(z0: np.ndarray,  port_idx: int) -> np.ndarray:
+    """
+    Calculate the load impedance for a given port index by parallel combining the impedances
+    of all other ports.
+
+    Parameters
+    ----------
+    z0 : ndarray
+        Port impedances, shape (n_frequencies, n_ports).
+    port_idx : int
+        The port index for which to calculate the load impedance.
+
+    Returns
+    -------
+    z0_load : number or None
+        the load impedance for the given port index, or None if the
+        media does not have a load impedance for that port index.
+
+    Raises
+    ------
+    IndexError
+        If the port index is out of range for the given port impedances.
+    """
+    nports: int = z0.shape[1]
+
+    if port_idx >= nports:
+        raise IndexError(f"Port index {port_idx} is out of range for Network, which has {nports} ports.")
+
+    if nports == 1:
+        return np.ones_like(z0) * INF
+
+    mask = np.arange(nports) != port_idx
+
+    return 1. / np.sum(1. / z0[:, mask], axis=1)
+
+
+def splitter_s(z0: np.ndarray) -> np.ndarray:
+    """
+    Generate ideal, lossless n-way splitter Network's s-parameters.
+
+    Parameters
+    ----------
+    z0 : :class:`numpy.ndarray`
+        Impedances of the ports
+
+    Returns
+    -------
+    s : :class:`numpy.ndarray`
+        S-parameters of the splitter with given impedances
+
+    See Also
+    --------
+    splitter : splitter generator method.
+    """
+    nports = z0.shape[1]
+    s = 2 * np.sqrt(np.einsum('ki,kj->kij', z0.real, z0.real)) / np.einsum('ki,kj->kij', z0, z0)
+    s /= np.sum(1. / z0, axis=1)[:, None, None]
+
+    ports_idx = np.arange(nports)
+    z0_load = np.array([get_z0_load(z0=z0, port_idx=i) for i in ports_idx]).T
+    s[:, ports_idx, ports_idx] = (z0_load - z0.conj()) / (z0_load + z0)
```

Comment:
doesn't the equation defined in the `splitter`'s docstring is (`z0 - z0_load.conj()`) instead ?